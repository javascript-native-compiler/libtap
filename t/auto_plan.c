#define USE_AUTO_PLAN
#include "tap.h"

void test_everything () {
    char buf1[4] = { 0, 1, 2, 3 };
    char buf2[4] = { 0, 1, 2, 3 };

    cmp_ok(1, "==", 1, "cmp_ok");
    ok(1, "ok");
    is("abc", "abc", "is");
    isnt("abc", "foo", "isnt");
    cmp_mem(buf1, buf2, sizeof(buf1), "cmp_mem");
    like("fnord", "f(yes|no)r*[a-f]$", "like");
}

int main () {
    plan(AUTO_PLAN);
    test_everything();
    done_testing();
}
