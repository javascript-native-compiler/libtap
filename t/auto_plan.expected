1..6
ok 1 - cmp_ok
ok 2 - ok
ok 3 - is
ok 4 - isnt
ok 5 - cmp_mem
ok 6 - like
